Corrected TV form with the ResourceList type

Ruslan-Aleev opened the pull request:
### What does it do?
Corrected the TV form with the ResourceList type:
- Changed the order
- Related items grouped
- Removed unused params in TV render in resource
- **Added list check via "Allow blank" instead of "Allow empty choice"**

Later I will correct for TV with other types.

**Before:**
![tv-res-1](https://user-images.githubusercontent.com/12523676/78137183-8f045e80-742d-11ea-9c42-e70dbce3fe57.png)

**After:**
![tv-res-2](https://user-images.githubusercontent.com/12523676/78147379-1658ce80-743c-11ea-914a-d51f24e9e391.png)

### Why is it needed?
Improves UI / UX
It became more beautiful :)

### Related issue(s)/PR(s)
https://github.com/modxcms/revolution/pull/15042
https://github.com/modxcms/revolution/pull/15009

Ruslan-Aleev committed:
Corrected TV form with the ResourceList type

JoshuaLuckers commented:
Would it make sense to remove the option "allow empty choice" and add an empty choice to the select list when "allow blank" is set to "no"?

Ruslan-Aleev commented:
Unfortunately, I do not understand how the "Allow empty choice" parameter is used in the rendering of the resource. It displays "-" at the beginning of the list, but I did not find where exactly this happens.

p.s. By the way, there are other parameters that are not displayed in the TV settings, but are checked in the render of the resource, for example - https://github.com/modxcms/revolution/blob/3.x/manager/templates/default/element/tv/renders/input/resourcelist.tpl#L21

Ruslan-Aleev committed:
Added list check via "Allow blank" instead of "Allow empty choice"

Ruslan-Aleev committed:
Removed unnecessary lexicons

Ruslan-Aleev commented:
@JoshuaLuckers I changed the PR, check.
I added the display "-" at the top of the list always, because otherwise, the first resource was put at the top of the TV list; and you could save this resource in TV without noticing, because allowBlank = true (I'm about https://github.com/modxcms/revolution/pull/15043/commits/a62db2fd5f2ebf2af80b4a5caf7363830cf684fe).

JoshuaLuckers reviewed (approved):
Great work!

Ruslan-Aleev committed:
Removed unused params in TV render in resource

JoshuaLuckers commented:
If this is approved and merged we need to update [the relevant documentation](https://docs.modx.com/3.x/en/building-sites/elements/template-variables/input-types).

Jako commented:
Typeahead would be nice to have there. Similar as the removed typeahead in #15044.
